sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -226,35 +226,21 @@ def __iter__(self):
     def _iter_contextual_wrapper(self, max_wait_time=None):
         """The purpose of this wrapper is to allow both state restoration (for multiple concurrent iteration)
         and per-iter argument passing that requires the former."""
-        # pylint: disable=protected-access
-        original_timeout = None
         while True:
-            # This is not threadsafe, but gives us a way to handle if someone passes
-            # different max_wait_times to different iterators and uses them in concert.
-            if max_wait_time:
-                # _timeout to _idle_timeout
-                original_timeout = self._handler._idle_timeout
-                self._handler._idle_timeout = max_wait_time * 1000
             try:
-                message = self._inner_next()
+                message = self._inner_next(wait_time=max_wait_time)
                 links = get_receive_links(message)
                 with receive_trace_context_manager(self, links=links):
                     yield message
             except StopIteration:
                 break
-            finally:
-                if original_timeout:
-                    try:
-                        self._handler._timeout = original_timeout
-                    except AttributeError:  # Handler may be disposed already.
-                        pass
-
-    def _inner_next(self):
+
+    def _inner_next(self, wait_time=None):
         # We do this weird wrapping such that an imperitive next() call, and a generator-based iter both trace sanely.
         self._check_live()
         while True:
             try:
-                return self._do_retryable_operation(self._iter_next)
+                return self._do_retryable_operation(self._iter_next, wait_time=wait_time)
```

Comment:
Is this going to wait the "max time" for every retry attempt?
The wait time a customer passes in needs to be the cumulative total, i.e. _actually_ the longest they can expect to wait.

Reply:
And there should probably be a test for the timing if there's not one already.

Reply:
This is just passing the max_wait_time for the user can set for the iterator through to the _iter_next function, it should not impact the timeout property that the user sets for retries via kwargs

Reply:
We also have this test here: test_queue_receiver_respects_max_wait_time_overrides to test max_wait_time on the iterator!